app/controllers/admin/divisions_controller.rb

```diff
@@ -0,0 +1,14 @@
+class Admin::DivisionsController < ApplicationController
+  before_action :authenticate_user!
+  after_action :verify_authorized
+
+  def select
+    redisplay_url = params[:redisplay_url] || root_path
+    division_id = params[:division_id]
+    set_selected_division_id(division_id)
+    division = Division.find_safe(division_id)
+    authorize division || current_division
```

Comment:
I'm confused here. division_id gets saved, but then division may be null if find_safe fails, in which case a _different_ division is authorized against? If find_safe fails, shouldn't we not be storing the errant division_id?